src/pymor/analyticalproblems/expressions.py

```diff
@@ -0,0 +1,307 @@
+# This file is part of the pyMOR project (http://www.pymor.org).
+# Copyright 2013-2020 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+from numbers import Number
+from itertools import zip_longest
+import numpy as np
+
+from pymor.parameters.base import ParametricObject
+
+
+builtin_max = max
+
+
+def parse_expression(expression, dim_domain, parameters={}, values={}):
+    if isinstance(expression, Expression):
+        return expression
+    locals_dict = {name: Parameter(name, dim) for name, dim in parameters.items()}
+    locals_dict['x'] = X(dim_domain)
+    return _convert_to_expression(eval(expression, dict(globals(), **values), locals_dict))
+
+
+class Expression(ParametricObject):
+
+    shape = None
+
+    def to_numpy(self):
+        expression = self.numpy_expr()
+        code = compile(expression, '<expression>', 'eval')
```

Comment:
Does this produce 'good' errors on malformed expressions?

Reply:
No. But there should never be any malformed expression. If there is, it's a bug in this module. Malformed expressions should be catched when the expression is created. (Of course, at the moment most error handling is still missing.)

Reply:
I did some work on error handling. Most malformed expression (when being parsed from a string) should now produce halfway useful error messages.

As before, errors in `to_numpy`/`numpy_expr` will be hard to understand for the user. But they should not happen in the first place.